crates/transpiler/src/target/mod.rs

```diff
@@ -891,6 +895,381 @@ impl Target {
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use qiskit_accelerate::target_transpiler::Target;
+    ///
+    /// let mut target = Target::new().with_pulse_alignment(4);
+    /// assert_eq!(target.pulse_alignment, 4)
+    /// ```
+    pub fn with_pulse_alignment(mut self, pulse_alignment: u32) -> Self {
+        self.pulse_alignment = pulse_alignment;
+        self
+    }
+
+    /// Adds an `acquire_alignment` value representing a time resolution of
+    /// measure instruction starting time for the [Target].
+    ///
+    /// **Warning:** Should only be used during the construction process of [Target],
+    /// after calling [Target::new] or any other construction methods.
+    ///
+    /// # Arguments
+    ///
+    /// * `acquire_alignment` - An integer value representing a time
+    ///        resolution of measure instruction starting time. Measure
+    ///        instruction should start at time which is a multiple of the
+    ///        alignment value.
+    ///  
+    /// # Returns
+    ///
+    /// Initialized [Target] with a defined ``acquire_alignment`` attribute.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use qiskit_accelerate::target_transpiler::Target;
+    ///
+    /// let mut target = Target::new().with_acquire_alignment(5);
+    /// assert_eq!(target.acquire_alignment, 5)
+    /// ```
+    pub fn with_acquire_alignment(mut self, acquire_alignment: u32) -> Self {
+        self.acquire_alignment = acquire_alignment;
+        self
+    }
+
+    /// Sets a list of the characteristics of each qubit on the [Target] device.
+    ///
+    /// **Warning:** Should only be used during the construction process of [Target],
+    /// after calling [Target::new] or any other construction methods.
+    ///
+    /// # Arguments
+    ///
+    /// * `qubit_properties` - A list of python `QubitProperties` objects defining
+    ///        the characteristics of each qubit on the target device. If specified
+    ///        the length of this list must match the number of qubits in the target,
+    ///        where the index in the list matches the qubit number the properties
+    ///        are defined for. If some qubits don't have properties available you
+    ///        can set that entry to ``None``.
+    ///
+    /// # Returns
+    ///
+    /// Initialized [Target] with a defined ``qubit_properties`` attribute.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if there was a specified [Target::num_qubits] attribute and the
+    /// the length of the provided list is not equal to [Target::num_qubits].
+    ///
+    /// /// # Examples
+    ///
+    /// ```rust
+    /// use qiskit_accelerate::target_transpiler::Target;
+    ///
+    /// let mut target = Target::new().with_qubit_properties(vec![]);
+    /// assert_eq!(target.num_qubits, Some(0));
+    /// ```
+    pub fn with_qubit_properties(self, qubit_properties: Vec<PyObject>) -> Self {
```

Comment:
Is this actually a thing we should have in the rust builder api? It doesn't seem like it's something we'll ever define in Rust while it's a `PyObject`.